version: '3.8'

services:
  zitadel:
    image: ghcr.io/zitadel/zitadel:stable
    container_name: zitadel
    command: start-from-init --config /zitadel-config.yml --config /zitadel-secrets.yml --steps /zitadel-init-steps.yml --masterkey "${ZITADEL_MASTERKEY}" --tlsMode external
    environment:
      - ZITADEL_FIRSTINSTANCE_ORG_HUMAN_USERNAME=Admin1!@elotoja.com
      - ZITADEL_FIRSTINSTANCE_ORG_HUMAN_PASSWORD=Password1!
    volumes:
      - ./zitadel-config.yml:/zitadel-config.yml:ro
      - ./zitadel-secrets.yml:/zitadel-secrets.yml:ro
      - ./zitadel-init-steps.yml:/zitadel-init-steps.yml:ro
    ports:
      - 8080:8080
    networks:
      - proxy
    labels:
      traefik.enable: true
      traefik.http.routers.zitadel.entryPoints: https
      traefik.http.routers.zitadel.rule: Host(`zitadel.${DOMAIN}`)
      traefik.http.services.zitadel.loadbalancer.server.scheme: h2c
      traefik.http.services.zitadel.loadbalancer.passHostHeader: true
      traefik.http.services.zitadel.loadbalancer.server.port: 8080
    restart: always

networks:
  proxy:
    external: true

